File: AdegaOak/AdegaOak.Models/Models/Movimentacao.cs
namespace AdegaOak.Models.Models;

public class Movimentacao
{
    public int Id { get; set; }
    public DateTime Data { get; set; } = DateTime.UtcNow;
    public string Tipo { get; set; } = string.Empty;       // "Entrada" | "Saída"
    public string TipoVenda { get; set; } = "Varejo";      // "Varejo" | "Atacado"
    public int ProdutoId { get; set; }
    public string ProdutoDescricao { get; set; } = string.Empty;
    public int Quantidade { get; set; }
    public int UsuarioId { get; set; }
    public string Responsavel { get; set; } = string.Empty;
    public string? TipoSaida { get; set; }                 // Consumo, Vencimento, Avaria, Combo, etc.
    public decimal ValorUnitario { get; set; }
    public decimal ValorTotal => ValorUnitario * Quantidade;

    // Navigation
    public Produto? Produto { get; set; }
    public Usuario? Usuario { get; set; }
}

File: AdegaOak/frontend/types/movimentacao.ts
export interface MovimentacaoDto {
  id: number;
  data: string;
  tipo: 'Entrada' | 'Saída';
  tipoVenda: string;
  produtoId: number;
  produtoDescricao: string;
  quantidade: number;
  usuarioId: number;
  responsavel: string;
  tipoSaida: string | null;
  valorUnitario: number;
  valorTotal: number;
}

export interface CreateMovimentacaoRequest {
  tipo: string;
  tipoVenda: string;
  produtoId: number;
  quantidade: number;
  tipoSaida?: string | null;
  valorUnitario: number;
}

export interface MovimentacaoFiltroRequest {
  tipo?: string;
  tipoVenda?: string;
  usuarioId?: number;
  dataInicio?: string;
  dataFim?: string;
  produtoId?: number;
}

export interface MovimentacaoResumoDto {
  totalEntradas: number;
  totalSaidas: number;
  saldo: number;
  quantidadeMovimentacoes: number;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}
